controllers/storagecluster/blackbox_exporter.go

```diff
@@ -0,0 +1,698 @@
+			Labels:    blackboxExporterLabels,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: blackboxExporterLabels,
+			Ports: []corev1.ServicePort{
+				{
+					Name:       portBlackboxHTTP,
+					Port:       9115,
+					TargetPort: intstr.FromInt(9115),
+					Protocol:   corev1.ProtocolTCP,
+				},
+			},
+			Type: corev1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+// createBlackboxServiceMonitor enables Prometheus scraping
+func (r *StorageClusterReconciler) createBlackboxServiceMonitor(ctx context.Context, instance *ocsv1.StorageCluster) error {
+	expected := r.getDesiredBlackboxServiceMonitor(instance)
+	actual := &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      expected.Name,
+			Namespace: expected.Namespace,
+		},
+	}
+
+	err := r.Client.Get(ctx, types.NamespacedName{Name: actual.Name, Namespace: actual.Namespace}, actual)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Create new ServiceMonitor
+			if err := controllerutil.SetControllerReference(instance, expected, r.Scheme); err != nil {
+				r.Log.Error(err, "Failed to set owner reference on ServiceMonitor", "ServiceMonitor", expected.Name)
+				return err
+			}
+			if err := r.Client.Create(ctx, expected); err != nil {
+				r.Log.Error(err, "Failed to create ServiceMonitor", "ServiceMonitor", expected.Name)
+				return err
+			}
+			r.Log.Info("Created ServiceMonitor", "ServiceMonitor", expected.Name)
+			return nil
+		}
+		r.Log.Error(err, "Failed to get ServiceMonitor", "ServiceMonitor", expected.Name)
+		return err
+	}
+
+	// Only update if our labels or spec differ
+	if !equality.Semantic.DeepEqual(expected.Labels, actual.Labels) ||
+		!equality.Semantic.DeepEqual(expected.Spec, actual.Spec) {
+
+		actual.Labels = expected.Labels
+		actual.Spec = expected.Spec
+
+		if err := controllerutil.SetControllerReference(instance, actual, r.Scheme); err != nil {
+			r.Log.Error(err, "Failed to set owner reference on ServiceMonitor", "ServiceMonitor", actual.Name)
+			return err
+		}
+		if err := r.Client.Update(ctx, actual); err != nil {
+			r.Log.Error(err, "Failed to update ServiceMonitor", "ServiceMonitor", actual.Name)
+			return err
+		}
+		r.Log.Info("Updated ServiceMonitor", "ServiceMonitor", actual.Name)
+	}
+
+	return nil
+}
+
+func (r *StorageClusterReconciler) getDesiredBlackboxServiceMonitor(instance *ocsv1.StorageCluster) *monitoringv1.ServiceMonitor {
+	sm := &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      blackboxExporterName,
+			Namespace: blackboxExporterNamespace,
+			Labels:    blackboxExporterLabels,
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			NamespaceSelector: monitoringv1.NamespaceSelector{MatchNames: []string{blackboxExporterNamespace}},
+			Selector:          metav1.LabelSelector{MatchLabels: blackboxExporterLabels},
+			Endpoints: []monitoringv1.Endpoint{
+				{
+					Port:     portBlackboxHTTP,
+					Path:     pathMetrics,
+					Interval: blackboxScrapeInterval,
+				},
```

Comment:
you need to define the icmp probe inside the service monitor to get latency between nodes